### PR TITLE
C sharp oop 2

### PR DESCRIPTION
**Pull request by 9inebar**
homework for C sharp oop lesson 2 part

### CODE REVIEW

File: CSharpOOP/Program.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        // Parsing
+        string name = item[0];
+        int count = Int32.Parse(item[1]);
+        double price = double.Parse(item[2], CultureInfo.InvariantCulture);
+        double discount = double.Parse(item[3], CultureInfo.InvariantCulture);
+        // Printing
+        PrintShoppingItem(name, price, discount, count);
+    }
+
+    string[] EnterShoppingItem()
```
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:28]: It is great thet you've created a separate method to read from console

File: CSharpOOP/Program.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            totalPrice);
+    }
+
+    #endregion
+
+    #region Task2
+
+    public void Task2()
+    {
+        //create array variable for ShoppingItem class constructor with a max number of 3 elements
```
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:55]: Great that you added comments that help understand execution of a program

File: CSharpOOP/Program.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            //Read 3 items
+            string[] item = EnterShoppingItem();
+            items[i] = new ShoppingItem(item[0], item[1], item[2], item[3]);
+        }
+
+        // print final receipt
+        printShopingItemArray(items);
+    }
+
+    void printShopingItemArray(ShoppingItem[] items)
```
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:68]: printShopingItemArray -> PrintShoppingItemArray

File: CSharpOOP/Program.cs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    {
+        Console.WriteLine("Receipt: ");
+        foreach (ShoppingItem item in items)
+        {
+            //print item
+            if (item.count == 0)
+            {
+                PrintShoppingItem(item.name, item.price, item.discount);
```
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:76]: Probably if count == 0  we can skip this item in final receipt

Comment (dnnyhlwdplc) [reply CSharpOOP/Program.cs:76]: But if your aim was to show method ececution with and without optional param - it is okay. 
But please doublecheck the logic :)

File: CSharpOOP/Program.cs [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+    #region Task3
+    public void Task3()
+    { 
+        Task2();
+    }
+    #endregion
+
+    #region Task4
+    public void calcPerimeter() //triangle perimeter calc method
```
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:97]: calcPerimeter() -> CalcPerimeter()

File: CSharpOOP/Program.cs [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+            Console.WriteLine("Type the geometry shape edge values");
+            int size = 0;
+            size = Convert.ToInt32(Console.ReadLine())!;
+            while (size <= 3)
+            {
+                Console.WriteLine($"To calculate geometry shape you need to have at least 3 sizes");
+                size = Convert.ToInt32(Console.ReadLine())!;
+            }
+
+            double[] Edges = new Double[Convert.ToInt32(Console.ReadLine())!];
```
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:108]: double[] Edges = new Double[size];

File: CSharpOOP/Program.cs [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+            double TrianglePerimeter = 0;
+            foreach (double Edge in Edges)
+            {
+                TrianglePerimeter += Edge;
+            }
+
+            Console.WriteLine($"Triangle perimeter is going to be {TrianglePerimeter}");
+
+            double calcPerimeter(params double[] sides)
```
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:123]: calcPerimeter -> CalcPerimeter

File: CSharpOOP/Program.cs [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+            double[] Edges = new Double[Convert.ToInt32(Console.ReadLine())!];
+            for (int i = 0; i < Edges.Length; i++)
+            {
+                Console.WriteLine($"Enter the edge {i + 1}");
+                Edges[i] = Convert.ToDouble(Console.ReadLine());
+            }
+
+            double TrianglePerimeter = 0;
+            foreach (double Edge in Edges)
```
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:116]: Reuse calcPerimeter(params double[] sides)

File: CSharpOOP/Program.cs (hunk 5 above)
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:97]: In task you should've called themethod for 3 figures :)

File: CSharpOOP/Program.cs [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+                {
+                    result += side;
+                }
+
+                return result;
+            }
+        }
+    #endregion
+
+    #region Task6
```
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:136]: You forgot about task 5 :(

File: CSharpOOP/Program.cs [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+    /*
+    Наследование:
+    Существует какая-то первая модель автомобиля 80 года выпуска,
```
Comment (dnnyhlwdplc) [inline CSharpOOP/Program.cs:139]: Nice examples!

File: CSharpOOP/ShoppingItem.cs [hunk 11]
```diff
@@ -0,0 +1,22 @@
+using System.Globalization;
+
+namespace CSharpOOP;
+
+public class ShoppingItem
```
Comment (dnnyhlwdplc) [inline CSharpOOP/ShoppingItem.cs:5]: Nice usage of new material!

File: CSharpOOP2/Program.cs [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+
+using CSharpOOP2;
+
+//first example of the car class
+Car myCar = new Car();
+myCar.color = "green";
+myCar.fuelLevel = 0;
+myCar.tankVolume = 80;
+Console.WriteLine("Enter the fuel amount");
+double FuelLiter = double.Parse(Console.ReadLine()!);
```
Comment (dnnyhlwdplc) [inline CSharpOOP2/Program.cs:11]: FuelLiter -> fuelLiter

File: CSharpOOP2/Program.cs [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+
+//internal car method call
+myCar.turnTheLightsOn();
+
+//fourth example of the car class
+Car myCar4 = new Car();
+myCar4.color = "white";
+myCar4.tankVolume = 70;
+double FuelLiter5 = double.Parse(Console.ReadLine()!);
+myCar4.FuelACar(FuelLiter5);
```
Comment (dnnyhlwdplc) [inline CSharpOOP2/Program.cs:39]: myCar4.FuelACar(FuelLiter5) and myCar4.FuelACar(FuelLiter6)  what is its purpose?

**Review by dnnyhlwdplc — CHANGES_REQUESTED** [review]

File: CSharpOOP/Program.cs (hunk 9 above)
Comment (9inebar) [reply CSharpOOP/Program.cs:136]: I didn't have time to finish it

File: CSharpOOP2/Program.cs (hunk 12 above)
Comment (9inebar) [reply CSharpOOP2/Program.cs:11]: sorry didn't get your comment

File: CSharpOOP2/Program.cs (hunk 13 above)
Comment (9inebar) [reply CSharpOOP2/Program.cs:39]: If we don't increment the iterations of inputs, the program will grab previous input results (variables)

File: CSharpOOP/Program.cs (hunk 3 above)
Comment (9inebar) [reply CSharpOOP/Program.cs:68]: could you please explain?

Comment (dnnyhlwdplc) [reply CSharpOOP/Program.cs:68]: fix naming :)

**Pull request merged**